### ISSUE TITLE
undefined symbol: btrfs_device_size with btrfs-progs 5.12

### ISSUE THREAD

**Issue by ava1ar** [openSUSE/snapper#650]
Upgrading to btrfs-progs 5.12 broke snapper on Arch:
```
snapper: symbol lookup error: /usr/lib/libbtrfs.so.0: undefined symbol: btrfs_device_size
```

Is this snapper issue or btrfs-progs issue?

**Comment by aschnell** [comment]
That is more likely a btrfs-progs issue. The symbol is not used inside snapper.

**Comment by kdave** [comment]
Yes it's a bug in btrfs-progs, I'll take a look.

**Comment by kdave** [comment]
This should fix it, I'll have to do a 5.12.1 release so it gets fixed properly.

```diff
--- a/Makefile
+++ b/Makefile
@@ -149,8 +149,7 @@ objects = kernel-shared/dir-item.o \
          common/string-table.o common/task-utils.o \
          kernel-shared/inode.o kernel-shared/file.o common/help.o cmds/receive-dump.o \
          common/fsfeatures.o \
-         common/format-output.o \
-         common/device-utils.o
+         common/format-output.o
 cmds_objects = cmds/subvolume.o cmds/filesystem.o cmds/device.o cmds/scrub.o \
               cmds/inspect.o cmds/balance.o cmds/send.o cmds/receive.o \
               cmds/quota.o cmds/qgroup.o cmds/replace.o check/main.o \
@@ -174,7 +173,7 @@ libbtrfs_objects = common/send-stream.o common/send-utils.o kernel-lib/rbtree.o
                   common/device-scan.o common/path-utils.o \
                   common/utils.o libbtrfsutil/subvolume.o libbtrfsutil/stubs.o \
                   crypto/hash.o crypto/xxhash.o $(CRYPTO_OBJECTS) \
-                  common/open-utils.o common/units.o
+                  common/open-utils.o common/units.o common/device-utils.o
 libbtrfs_headers = common/send-stream.h common/send-utils.h send.h kernel-lib/rbtree.h btrfs-list.h \
               crypto/crc32c.h kernel-lib/list.h kerncompat.h \
               kernel-lib/radix-tree.h kernel-lib/sizes.h kernel-lib/raid56.h \```

**Comment by aschnell** [comment]
Will be fixed in btrfs-progs.
